content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+---
+authors: ["carmine-laface"]
+date: "2020-06-23"
+draft: false
+share: true
+categories: [English, Software Testing]
+title: "Automation testing: a step back"
+type: "post"
+languageCode: "en-EN"
+toc: true
+---
+
+The goal of this article is to define basic concepts related to testing, trying not to take anything for granted.
+
+# Why we test?
+
+Why is it important to write automated tests? I asked myself and I did some research because the answer to this question was not obvious to me. I knew it was important but I didn't know why. So I decided to try to motivate it starting from a point of view as impartial as possible.
+
+After some research I came to the conclusion that the benefits of the tests are four: they can avoid regression, they can count as documentation, they can help write better code and they can help find bugs during development.
+
+Let's focus on one of them at the time and analyze it.
+
+## Avoid regression
+
+> When you implemented a feature in your app, you likely ran the code to verify that it worked as expected. You performed a test, albeit a manual test. As you continued to add and update features, you probably also continued to run your code and verify it works. But doing this manually every time is tiring, prone to mistakes, and does not scale.  
+Computers are great at scaling and automation! So developers at companies large and small write automated tests, which are tests that are run by software and do not require you to manually operate the app to verify the code works.
+
+This is the introduction of a [Google test codelab](https://codelabs.developers.google.com/codelabs/advanced-android-kotlin-training-testing-basics/#0). It not only defines in a pretty straightforward way what automated tests are, but it also tells why you should do that. The reason mentioned here is only the first of the four on the list that I have compiled before, but this is the main reason.
+
+So we write automated tests to save time in the future, it's an investment. We suppose that when we add more features the old ones need to be checked again, and we invest our time in order to save more in the future, but in the short term we will not have a return.
```

Comment:
So, we write automated tests to save time in the future, it's an investment. We suppose that when we add more features the old ones need to be checked again. We invest our time in order to save more in the future but in the short term we will not have a return.